extras/minimize_testcase.cpp

```diff
@@ -0,0 +1,251 @@
+  return std::make_pair(outer, i);
+}
+
+void Dump(const Polygons &polys) {
+  for (auto poly : polys) {
+    std::cout << "polys.push_back({" << std::setprecision(9) << std::endl;
+    for (auto v : poly) {
+      std::cout << "    {" << v.x << ", " << v.y << "},  //" << std::endl;
+    }
+    std::cout << "});" << std::endl;
+  }
+  for (auto poly : polys) {
+    std::cout << "show(array([" << std::endl;
+    for (auto v : poly) {
+      std::cout << "  [" << v.x << ", " << v.y << "]," << std::endl;
+    }
+    std::cout << "]))" << std::endl;
+  }
+}
+
+void DumpTriangulation(const Polygons &polys, float precision) {
+  ExecutionParams oldParams = PolygonParams();
+  manifold::PolygonParams().processOverlaps = true;
+  auto result = Triangulate(polys);
+  PolygonParams() = oldParams;
+  for (auto &tri : result) {
+    auto xid = findIndex(polys, tri.x);
+    auto yid = findIndex(polys, tri.y);
+    auto zid = findIndex(polys, tri.z);
+    auto x = polys[xid.first][xid.second];
+    auto y = polys[yid.first][yid.second];
+    auto z = polys[zid.first][zid.second];
+    printf("show(array([[%.7f, %.7f], [%.7f, %.7f], [%.7f, %.7f]]))\n", x.x,
+           x.y, y.x, y.y, z.x, z.y);
+    if (!CCW(x, y, z, precision)) {
+      printf("^ not CCW\n");
+    }
+  }
+}
+
+// x direction ray
+// idk which direction, but it is a fixed direction...
+bool rayHit(glm::vec2 point, glm::vec2 q0, glm::vec2 q1) {
+  return ((q0.y >= point.y) != (q1.y >= point.y)) &&
+         (point.x <= (q1.x - q0.x) * (point.y - q0.y) / (q1.y - q0.y) + q0.x);
+}
+
+std::vector<int> getChildren(Polygons &polys, int i) {
+  std::vector<int> results;
+  for (int j = 0; j < polys.size(); j++) {
+    if (i == j) continue;
+    glm::vec2 point = polys[j][0];
+    auto k1 = [&](size_t k) {
+      return k == (polys[i].size() - 1) ? 0 : (k + 1);
+    };
+    int count = std::count_if(
+#if MANIFOLD_PAR == 'T' && !(__APPLE__)
+        std::execution::par,
+#endif
+        countAt(0ul), countAt(polys[i].size()),
+        [&](size_t k) { return rayHit(point, polys[i][k], polys[i][k1(k)]); });
+    if (count % 2) results.push_back(j);
+  }
+  return results;
+}
+
+// we are assuming polys is valid
+// we try to remove vertices from polys such that
+// 1. the updated polys is still valid (no overlapping edges)
+// 2. error in triangulation (either geometryErr or overlapping triangles)
+void simplify(Polygons &polys, float precision = -1) {
+  bool removedSomething = true;
+  std::string msg;
+  while (removedSomething) {
+    int points = 0;
+    for (auto &poly : polys) points += poly.size();
+    std::cout << "trying to simplify " << points << " points" << std::endl;
+
+    removedSomething = false;
+    // try to remove simple polygons
+    for (int i = 0; i < polys.size(); i++) {
+      std::vector<int> children = getChildren(polys, i);
+      // if there are children, we can't remove it or we will mess up with
+      // winding direction...
```

Comment:
I'll have to study your algorithm sometime when I'm less tired, but I trust you. This will be a great convenience function even if it misses an edge case or two (not that I've found any).

Reply:
I can add more comments to explain it as well.